### PR TITLE
Update lxd code paths to always return a traced error.

### PR DESCRIPTION
**Pull request by jameinel**
## Description of change

Also remove all of the "+build go1.3" lines as we haven't supported <1.3
in a long time.

## QA steps

Nothing concrete should have been changed.

## Documentation changes

None.

## Bug reference

None.

### CODE REVIEW

**Comment by jameinel** [comment]
[Update LXD code paths to trace errors](https://trello.com/c/Oy5xeZQT/222-update-lxd-code-paths-to-trace-errors)

File: tools/lxdclient/client_network.go [hunk 1]
```diff
@@ -103,12 +101,12 @@ func CreateDefaultBridgeInDefaultProfile(client creator) error {
...
 	config, _, err := client.GetProfile("default")
 	if err != nil {
-		return err
+		return errors.Trace(err)
 	}
 	_, ok := config.Devices["eth0"]
 	if ok {
 		/* don't configure an eth0 if it already exists */
-		return nil
+		return errors.Trace(err)
```
Comment (manadart) [inline tools/lxdclient/client_network.go]: We know that err will be nil here, so it doesn't matter, but I would be inclined to leave it as a nil return.

**Comment by manadart** [comment]
Looks like there are indeed some places where specific error content is being anticipated.

File: tools/lxdclient/client_network.go (hunk 1 above)
Comment (jameinel) [reply tools/lxdclient/client_network.go]: yeah, just overzealous copy & paste

**Comment by jameinel** [comment]
credentials.go had a couple of "if os.IsNotExist()" checks that can be wrapped into "if os.IsNotExist(errors.Cause(err))" I think.
Will try it again.

**Comment by jameinel** [comment]
$$merge$$

**Pull request merged**